fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryAnalyzer.java

```diff
@@ -276,7 +277,13 @@ private Relation resolveTableRef(Relation relation, Scope scope, Set<TableName>
                     }
 
                     if (table.isSupported()) {
-                        tableRelation.setTable(table);
+                        if (table.isOlapTable()) {
+                            // Copying the olap table meta to avoid the lock when query
+                            Table copied = DeepCopy.copyWithGson(table, OlapTable.class);
```

Comment:
If you failed multi times to replan, I think you need to degeneration to lock to get rid of high concurrency load.

Reply:
Why only copy olapTable？may use other type table direct without lock?